pkg/controller/controller.go

```diff
@@ -47,55 +52,95 @@ var knownKinds = []knownKind{{
 
 // Workload represents a workload in the cluster. It contains the top level object and all of the pods.
 type Workload struct {
-	TopController unstructured.Unstructured
-	Pods          []unstructured.Unstructured
+	TopController   unstructured.Unstructured
+	Pods            []unstructured.Unstructured
+	PodSpec         *corev1.PodSpec
+	PodCount        int
+	RunningPodCount int
 }
 
-func getAllPods(ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, namespace string) ([]unstructured.Unstructured, error) {
+// Client is used to interact with the Kubernetes API
+type Client struct {
```

Comment:
non-blocking: I believe this breaks the entire external api usage, is that correct? If so, we probably want to document somehow. Although I'm realizing we don't really have documentation now :-)

Reply:
Yup, definitely breaks the current API.

Good point on docs...will follow up with some real documentation

Reply:
Added a full example in the README. Would be nice to get a godoc link eventually